text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+resolution to fill in missing `builtin` packages.
+
+↩ [*Proposal*][proposal]
+
+## Why permit patching of the standard library dependencies on nightly?
+[rationale-patching]: #why-permit-patching-of-the-standard-library-dependencies-on-nightly
+
+Being able to patch `builtin = true` dependencies and replace their source with
+a `path` dependency is required to be able to replace `rustc_dep_of_std`. As
+crates which use these sources cannot be published to crates.io, this would not
+enable a usable general-purpose mechanism for crates to modify the standard
+library sources. This capability is restricted to nightly as that is all that is
+required for it to be used in replacing `rustc_dep_of_std`.
+
+↩ [*Patches*][patches]
+
+## Why limit enabling standard library features to nightly?
+[rationale-features]: #why-limit-enabling-standard-library-features-to-nightly
+
+If it were possible to enable features of the standard library crates on stable
+then all of the standard library's current features would immediately be held to
+the same stability guarantees as the rest of the standard library, which is not
+desirable. See
+[*Allow enabling/disabling features with build-std*][future-features]
+
+↩ [*Features*][features]
+
+## Why default to public for the implicit standard library dependencies?
+[rationale-implicit-public]: #why-default-to-public-for-the-implicit-standard-library-dependencies
+
+There are crates building on stable which re-export from the standard library.
+If the implicit standard library dependency were not public then these crates
+would start to trigger the `exported_private_dependencies` lint when upgrading
+to a version of Cargo with an implicit standard library dependency.
+
+↩ [*Public and private dependencies*][public-and-private-dependencies]
+
+## Why follow the default privacy of explicit standard library dependencies?
+[rationale-explicit-private]: #why-follow-the-default-privacy-of-explicit-standard-library-dependencies
+
+This may be unintuitive when a user first writes an explicit standard library
+dependency, triggering the `exported_private_dependency` lint, but this would be
+caught immediately by the user. However, it is also unintuitive that the default
+for privacy of a explicitly written dependency would depend on which crate the
+dependency was (i.e. the standard library has a different default than
+everything else).
+
+↩ [*Public and private dependencies*][public-and-private-dependencies]
+
+## Why not support explicit standard library dependencies in `build-dependencies`?
+[rationale-no-deps-in-build-deps]: #why-not-support-explicit-standard-library-dependencies-in-build-dependencies
+
+`build-dependencies` only apply to build scripts which are run on the host
+toolchain. There is little advantage to using a custom standard library with
+build scripts as they are not part of the final output artifact and anywhere
+they can run already has a toolchain with host tools and a pre-built standard
+library.
```

Comment:
This feature isn't as much about build-std but about explicitly declaring std dependencies.

However, there is little space for a `no_std` build script.

However, I do worry about the inconsistency

Reply:
I don't mind having `build-dependencies` allow explicit std dependencies, we can permit that if we think the inconsistency is a problem.

Reply:
If you decide to move forward with it as-is, I would as
- Explicitly call out the trade off of lack of consistency
- Link out to this in the Unresolved Questions to help draw attention to it for the Cargo team to more discuss during the RFC process

Reply:
I've drawn attention to this in https://github.com/davidtwco/rfcs/pull/2/commits/1e6184b78b2834565b1167831ed06d6dd9cca027

Reply:
Changed the wording of this slightly in 2a330da09efc33164ff706d129a3bda4d1c43982